### DISCUSSION TITLE
Final Status Update (Week 6)

### DISCUSSION THREAD

**Discussion by roryeiffe** [roryeiffe/Fresh-Results#61] [General]
Respond to this thread with everything you've worked on from 4/22/21-end of the Spring 2021 semester.

What did you work on this week?

What sort of technical blockers are you facing?

What sort of new technologies did you learn, if any?

Put any links to commits/other contributions here.

Where would you like to see this project go in the future?

Feel free to add more questions as you think of them!

**Comment by roryeiffe** [comment]
## What did you work on this week?

This week, I worked on integrating the custom words with the content script. Now, as users type in custom words, those words are then filtered on the web page. The custom words are also stored so when the extension is closed and reopened, the changes are still there. I also presented the UI portion of our final presentation. 

## What sort of technical blockers are you facing?

No technical blockers at the moment. 

## What sort of new technologies did you learn, if any?

I learned how to use chrome local storage. I mostly went off of the code that my teammates had already written and tried to document things as I went. 

Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/issues/49
https://github.com/roryeiffe/Spoiler-Block/commit/f4334881b4f4a284162702449cea4b63b9d356bf
https://github.com/roryeiffe/Spoiler-Block/commit/f118739486da79775b231c89f06ed4a03a385f66
https://github.com/roryeiffe/Spoiler-Block/commit/aa0dd39dfdd7a32054182b922cefb51db8d29c70
https://github.com/roryeiffe/Spoiler-Block/commit/e8e6dce4ecbdf21c3fdbce6bd31d0f023959f7e1
https://github.com/roryeiffe/Spoiler-Block/commit/4aa6a3002b620053f198a718f261c5f37a05203e
https://github.com/roryeiffe/Spoiler-Block/commit/647cc9da2cde4da180875889e5d36a46f238f391

Where would you like to see this project go in the future?

I would like to see this extension expanded to other browsers. It would also be really cool to see contributions from non-RPI students. Finally, I would like to see the project be made publicly available on web stores.

**Comment by SerenaChen1** [comment]
## What did you work on this week?

This week, I worked on fixing a bug that I caught, where we were saving null values for the color, which caused the spoiler censors to be colorless/transparent, defeating the whole purpose of them. 

## What sort of technical blockers are you facing?

There ended up being a lot of merge conflicts because Abdul had reorganized our file structure, but he helped resolve the ones I couldn't figure out. After my code got merged in, I noticed a small bug where the censors weren't showing up on load of a webpage, so I let him know, and he fixed it quickly. These technical blockers were resolved quickly.

## What sort of new technologies did you learn, if any?

I didn't learn any new technologies this week.

## Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/commit/9946a4602c736220df3e112e5eb07d2d85773f70
https://github.com/roryeiffe/Spoiler-Block/issues/56
https://github.com/roryeiffe/Spoiler-Block/pull/57 

## Where would you like to see this project go in the future?

Professor Turner mentioned in our presentation that since its open source, it should be relatively easy to have people working on the extension for multiple browsers simultaneously, and I think this is a great idea. I would also like to see a more advanced product, where we're not just filtering out all general keywords but rather specific ones for shows/films that the user enters. This was discussed in our early meetings, but we figured it was out of our scope, so we didn't follow through.

**Comment by VictoriaSavage526** [comment]
What did you work on this week?
I worked on editing some documentation to help people that want to help on the project understand the process

What sort of technical blockers are you facing?
N/A
What sort of new technologies did you learn, if any?
N/A

Put any links to commits/other contributions here.
[doc](https://github.com/roryeiffe/Spoiler-Block/commit/49338396dee9a8c82707ca85a78a7463ed0d327e)

Where would you like to see this project go in the future?
I would like to see additional functionality added cross-platform for different browsers, and maintain this so it doesn't gather too much dust.
Feel free to add more questions as you think of them!

**Comment by BreadGarLick** [comment]
What did you work on this week?

I worked on finishing up the autocomplete implementation with the new build and manifest

What sort of technical blockers are you facing?

I was unfamiliar with the new build and manifest, but got past it quickly.

What sort of new technologies did you learn, if any?

Nothing new really, just familiarizing myself with the new stuff

Put any links to commits/other contributions here.
793d946ecc5f6c3c21c11ac7f6e453fdd11cddb4
830d4336a5e916488067ce3fa54f499ab7ca9ed1
0adf7f83a3337fbfb6a8735dfd1ebc5c0c4aa15a
4556472de12c5b4a78a38742cbc4d7d78b067dbd
